lib/interval/interval.ex

```diff
@@ -89,12 +92,26 @@ defmodule Timex.Interval do
     case until do
       {:error, _} = err -> err
       _ ->
-        %__MODULE__{from: from, until: until,
-                    left_open: left_open, right_open: right_open,
-                    step: step}
+        case validate_step(step) do
+          :ok ->
+            %__MODULE__{from: from, until: until,
+                        left_open: left_open, right_open: right_open,
+                        step: step}
+          err -> err
+        end
```

Comment:
So I'll be he first person to say that I'm not in love with this nested case solution to this problem. Was attempting to chain a new validate_until and validate_step together, but it wasn't any more elegant than this. Any thoughts?